test_suite/tests/test_annotations.rs

```diff
@@ -811,6 +811,77 @@ fn test_serialize_with_enum() {
     );
 }
 
+#[derive(Debug, PartialEq, Serialize)]
+enum SerializeWithVariant {
+    #[serde(serialize_with="serialize_unit_variant_as_i8")]
+    Unit,
+
+    #[serde(serialize_with="SerializeWith::serialize_with")]
+    Newtype(i32),
+
+    #[serde(serialize_with="serialize_some_other_variant")]
+    Tuple(String, #[serde(skip_serializing_if="is_zero")] u8),
+
+    #[serde(serialize_with="serialize_some_other_variant")]
+    Struct {
+        f1: String,
+        #[serde(skip_serializing_if="is_zero")]
```

Comment:
Sure, I'll rip that part out. I was split on this and ended up erring on the side of compatibility with existing features.